engine/table/src/main/java/io/deephaven/engine/table/impl/ZeroKeyUpdateByManager.java

```diff
@@ -0,0 +1,102 @@
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.LinkedList;
+import java.util.Map;
+
+public class ZeroKeyUpdateByManager extends UpdateBy {
+    /** The output table for this UpdateBy operation */
+    final QueryTable result;
+
+    // this manager has only one bucket, managed by this object
+    final UpdateByBucketHelper zeroKeyUpdateBy;
+
+    /**
+     * Perform an updateBy without any key columns.
+     *
+     * @param description the operation description
+     * @param operators the operations to perform
+     * @param windows the unique windows for this UpdateBy
+     * @param inputSources the primitive input sources
+     * @param operatorInputSourceSlots maps the operators to source indices
+     * @param source the source table
+     * @param resultSources the result sources
+     * @param timestampColumnName the column to use for all time-aware operators
+     * @param redirHelper the row redirection helper for dense output sources
+     * @param control the control object.
+     */
+    protected ZeroKeyUpdateByManager(
+            @NotNull final String description,
+            @NotNull UpdateByOperator[] operators,
+            @NotNull UpdateByWindow[] windows,
+            @NotNull ColumnSource<?>[] inputSources,
+            @NotNull int[][] operatorInputSourceSlots,
+            @NotNull QueryTable source,
+            @NotNull final Map<String, ? extends ColumnSource<?>> resultSources,
+            @Nullable String timestampColumnName,
+            @NotNull UpdateByRedirectionHelper redirHelper,
+            @NotNull UpdateByControl control) {
+        super(source, operators, windows, inputSources, operatorInputSourceSlots, timestampColumnName, redirHelper,
+                control);
+
+        if (source.isRefreshing()) {
+            result = new QueryTable(source.getRowSet(), resultSources);
+
+            // this is a refreshing source, we will need a listener
+            listener = newUpdateByListener(description);
+            source.addUpdateListener(listener);
+            // result will depend on listener
+            result.addParentReference(listener);
+
+            // create input and output modified column sets
+            for (UpdateByOperator op : operators) {
+                op.createInputModifiedColumnSet(source);
+                op.createOutputModifiedColumnSet(result);
+            }
+
+            // create an updateby bucket instance directly from the source table
+            zeroKeyUpdateBy = new UpdateByBucketHelper(description, source, operators, windows, inputSources,
+                    operatorInputSourceSlots, resultSources, timestampColumnName, redirHelper, control);
+            buckets.offer(zeroKeyUpdateBy);
+
+            // make the source->result transformer
+            transformer = source.newModifiedColumnSetTransformer(result, source.getDefinition().getColumnNamesArray());
```

Comment:
There should be some pruning of the column names array argument, if the result columns overwrite any of the input columns. Applies to the bucketed case, too.